ui/server.py

```diff
@@ -417,6 +424,63 @@ def update_discovery_group():
         return 'OK', 200
 
 
+def assign_categories(discoveries):
+    """ Add the category to each discovery
+
+    Parameters
+    ----------
+    discoveries: list
+        List of discoveries without assigned categories to them
+
+    Returns
+    -------
+    list
+        List of discoveries with categories assigned to them
+    """
+    rulesdict, cat = _get_rules()
+    for discovery in discoveries:
+        if discovery['rule_id']:
+            discovery['category'] = rulesdict[discovery['rule_id']]['category']
+        else:
+            discovery['category'] = '(rule deleted)'
+
+    return discoveries
```

Comment:
I think this is not really needed, since lists and dictionaries are data structures managed by reference. So, if you add a new keypair in the function, it should also update the main data structure in the "main" method (e.g., the sort function)